lib/src/components/NxDateInput/stateful/NxStatefulDateInput.tsx

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { forwardRef } from 'react';
+
+import NxStatefulTextInput, {
+  Props as NxStatefulTextInputProps
+} from '../../NxTextInput/stateful/NxStatefulTextInput';
+
+import { Props, propTypes } from './types';
+export { Props, propTypes } from './types';
+
+const NxStatefulDateInput = forwardRef<HTMLDivElement, Props>(
+    (props, ref) => {
+      const typeDateProp = {
+        type: 'date' as NxStatefulTextInputProps['type']
```

Comment:
This shouldn't be needed.  I think the issue is that `NxStatefulTextInput` should be using `NxTextInputType` as its type, not `PublicNxTextInputType`.

Reply:
> This shouldn't be needed. I think the issue is that `NxStatefulTextInput` should be using `NxTextInputType` as its type, not `PublicNxTextInputType`.

Fixed: ad32bb0c1fc747913ee8a8bc8511d10b91906867

Reply:
> This shouldn't be needed. I think the issue is that `NxStatefulTextInput` should be using `NxTextInputType` as its type, not `PublicNxTextInputType`.

Removed the assertion here: 0ea12a39ad072b226c18b675a608f0cbd9821ee8